scripts/test/spec_helper.rb

```diff
@@ -0,0 +1,180 @@
+    cmd = command(arg)
+    while cmd.exit_status != 0 or cmd.stdout == '' and opt[:timeout] > 0
+      sleep 5
+      opt[:timeout] -= 5
+      cmd = command(arg)
+    end
+    return cmd
+  end
+
+  def curl_json_api(url)
+    output = cmd_with_retry("curl #{url}").stdout
+    if output.size > 0
+      JSON.parse(output)
+    else
+      {}
+    end
+  end
+
+  def load_json(file)
+    file = File.expand_path file
+    raise ArgumentError, "Invalid json file path: #{file}" unless File.exist? file
+    JSON.parse(gsub_env(File.read file))
+  end
+
+  def load_yaml(file)
+    file = File.expand_path file
+    raise ArgumentError, "Invalid json file path: #{file}" unless File.exist? file
+    YAML.load(gsub_env(File.read file))
+  end
+
+  def gsub_env(content)
+    content.gsub(/\$([a-zA-Z_]+[a-zA-Z0-9_]*)|\$\{(.+)\}/) { ENV[$1 || $2] }
+  end
+
+  def self.examples
+    Pathname.new(File.expand_path(File.join(__FILE__,'../../../examples')))
+  end
+
+  def self.tasks
+    if ENV["TASK"] != ""
+      pattern="#{examples}/tasks/#{ENV["TASK"]}"
+    else
+      pattern="#{examples}/tasks/*.y{a,}ml"
+    end
+    Dir.glob(pattern).collect{|f| File.basename f}
+  end
+
+  def add_plugins(plugins, type)
+    plugins.flatten.compact.uniq.each do |name|
+      @plugins.add([type, name])
+    end
+  end
+
+  def parse_task(t)
+    t.extend Hashie::Extensions::DeepFetch
+    t.extend Hashie::Extensions::DeepFind
+
+    m = t.deep_fetch("workflow", "collect", "metrics"){ |k| {} }
+    collectors = m.collect do |k, v|
+      k.match(/^\/intel\/(.*?)\/(.*?)/)
+      # NOTE: procfs/* doesn't follow the convention, nor does disk/smart.
+      if $1 == 'procfs' || $1 == 'disk'
+        case $2
+        when 'iface'
+          'interface'
+        when 'filesystem'
+          'df'
+        else
+          $2
+        end
+      else
+        $1
+      end
+    end
+    add_plugins(collectors, 'collector')
+
+    p = t.deep_find_all("process") || {}
+    processors = p.collect do |i|
+      if i.is_a? ::Array
+        i.collect{|j| j["plugin_name"] if j.include? "plugin_name"}
+      end
+    end
+    add_plugins(processors, 'processor')
+
+    p = t.deep_find_all("publish") || {}
+    publishers = p.collect do |i|
+      if i.is_a? ::Array
+        i.collect{|j| j["plugin_name"] if j.include? "plugin_name"}
+      end
+    end
+    add_plugins(publishers, 'publisher')
+  end
+
+  def plugins
+    @plugins ||= load_tasks
+  end
+
+  def load_tasks
+    @plugins = Set.new
+    SnapUtils.tasks.each do |t|
+      y = load_yaml SnapUtils.examples/"tasks/#{t}"
+      parse_task(y)
+    end
+    @plugins
+  end
+
+  def load_all_plugins
+    plugins.each do |i|
+      type, name = i
+      load_plugin(type, name)
+    end
+  end
+end
+
+RSpec.configure do |c|
+  c.formatter = 'documentation'
+  c.mock_framework = :rspec
+  c.verbose_retry = true
+  c.order = 'default'
+  c.include SnapUtils
+  if ENV["DEMO"] == "true" then
```

Comment:
It would be nice to add the DEMO mode consistently across all the plugins using the new large test framework.